packages/workbox-routing/lib/Router.mjs

```diff
@@ -0,0 +1,328 @@
+   * @return {Object} Returns an object with `handler` and `params` properties.
+   * They are populated if a matching route was found or `undefined` otherwise.
+   * @private
+   */
+  _findHandlerAndParams(event, url) {
+    const routes = this._routes.get(event.request.method) || [];
+    for (const route of routes) {
+      let matchResult = route.match({url, event});
+      if (matchResult) {
+        _private.logger.log(`The router found a matching route.`, {
+          route,
+          request: event.request,
+        });
+
+        if (Array.isArray(matchResult) && matchResult.length === 0) {
+          // Instead of passing an empty array in as params, use undefined.
+          matchResult = undefined;
+        } else if (matchResult.constructor === Object &&
+          Object.keys(matchResult).length === 0) {
+          // Instead of passing an empty object in as params, use undefined.
+          matchResult = undefined;
+        }
+
+        // Break out of the loop and return the appropriate values as soon as
+        // we have a match.
+        return {
+          params: matchResult,
+          handler: route.handler,
+        };
+      }
+    }
+
+    // If we didn't have a match, then return undefined values.
+    return {handler: undefined, params: undefined};
+  }
+
+  /**
+   * An optional `handler` that's called by default when no routes
+   * explicitly match the incoming request.
+   *
+   * If the default is not provided, unmatched requests will go against the
+   * network as if there were no service worker present.
+   *
+   * @param {function|module:workbox-runtime-caching.Handler} handler
+   * This parameter can be either a function or an object which is a subclass
+   * of `Handler`.
+   *
+   * Either option should result in a `Response` that the `Route` can use to
+   * handle the `fetch` event.
+   *
+   * See [handlerCallback]{@link module:workbox-routing.Route~handlerCallback}
+   * for full details on using a callback function as the `handler`.
+   */
+  setDefaultHandler(handler) {
+    this.defaultHandler = normalizeHandler(handler);
+  }
+
+  /**
+   * If a Route throws an error while handling a request, this `handler`
+   * will be called and given a chance to provide a response.
+   *
+   * @param {function|module:workbox-runtime-caching.Handler} handler
+   * This parameter can be either a function or an object which is a subclass
+   * of `Handler`.
+   *
+   * Either option should result in a `Response` that the `Route` can use to
+   * handle the `fetch` event.
+   *
+   * See [handlerCallback]{@link module:workbox-routing.Route~handlerCallback}
+   * for full details on using a callback function as the `handler`.
+   */
+  setCatchHandler(handler) {
+    this.catchHandler = normalizeHandler(handler);
+  }
+
+  /**
+   * Registers an array of routes with the router.
+   *
+   * @param {Array<module:workbox-routing.Route>} routes An array of routes to
+   * register.
+   */
+  registerRoutes(routes) {
```

Comment:
Is there any way we could see if anyone is using the registerRoutes method vs just using the singular registerRoute method.

Just think of code size.

Reply:
If you'd like to drop the multiple use case, we can do it now that we're making breaking changes.

Reply:
https://github.com/GoogleChrome/workbox/issues/856